extensions/content-manager/admin/src/components/SettingsViewWrapper/index.js

```diff
@@ -0,0 +1,329 @@
+      if(type === 'relation' && relationType === 'oneWay' ) {
+        return attr
+      }
+    });
+    if(!attributes[relationName]) {
+      return ['', ...options];
+    }
+    return options;
+  }
+
+  const getSelectOptions = input => {
+    if (input.name === 'settings.defaultSortBy' || input.name === 'settings.relationKey') {
+      return [
+        'id',
+        ...getListDisplayedFields().filter(
+          name =>
+            get(getAttributes, [name, 'type'], '') !== 'media' &&
+            name !== 'id' &&
+            get(getAttributes, [name, 'type'], '') !== 'richtext'
+        ),
+      ];
+    }
+
+    if (input.name === 'settings.mainField') {
+      const attributes = getAttributes;
+      const options = Object.keys(attributes).filter(attr => {
+        const type = get(attributes, [attr, 'type'], '');
+
+        return (
+          ![
+            'json',
+            'text',
+            'relation',
+            'component',
+            'boolean',
+            'date',
+            'media',
+            'richtext',
+          ].includes(type) && !!type
+        );
+      });
+      return options;
+    }
+
+    if (input.name === 'settings.relationName') {
+      const options =  getRelationNameOptions()
+      return options;
+    }
+
+    return input.options;
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    toggleWarningSubmit();
+    emitEvent('willSaveContentTypeLayout');
+  };
+
+  if (isLoading) {
+    return <LoadingIndicatorPage />;
+  }
+
+  const filterInputsByRelation = () => {
+    const filteredInputs = inputs.filter(input => {
+      return input.name !== 'settings.relationKey' && input.name !== 'settings.relationName'
+    });
+    const relationNameOptions = getRelationNameOptions();
+    if(modifiedData.settings && relationNameOptions.length) {
+      return modifiedData.settings.relation
+        ? inputs 
+        : filteredInputs
+    }
+    return filteredInputs;
+  }
+
+  const generateInputValue = (input) => {
+    if(input.name === 'settings.relation'){
+      const relationNameOptions = getRelationNameOptions();
+      return relationNameOptions.length > 0 ? get(modifiedData, input.name, '') : false;
+    }
+    return get(modifiedData, input.name, '')
+  }
+
+  return (
+    <>
+      <BackHeader onClick={goBack} />
+      <Container className="container-fluid">
+        <form onSubmit={handleSubmit}>
+          <Header {...headerProps} />
+          <div
+            className="row"
+            style={{
+              paddingTop: '3px',
+            }}
+          >
+            <Block
+              style={{
+                marginBottom: '13px',
+                paddingBottom: '30px',
+                paddingTop: '24px',
+              }}
+            >
+              <SectionTitle isSettings />
+              <div className="row">
+                {filterInputsByRelation().map(input => {
+                  return (
```

Comment:
You can remove return keyword and use `() => <Component />`

Reply:
fixed